tools/mo/openvino/tools/mo/utils/ir_engine/ir_engine.py

```diff
@@ -224,22 +223,48 @@ def __load_layer(self, layer):
                 layer_attrs.update(new_attrs)
             elif attr.tag == 'input':
                 inputs_counter = len(attr)
+
+                input = attr
+                for port in input:
+                    port_id = int(port.attrib['id'])
+                    input_shape = []
+                    port_rt_info = {}
+                    for dim in port:
+                        if dim.tag == "dim":
+                            input_shape.append(int(dim.text))
+                        if dim.tag == 'rt_info':
+                            for attr in dim:
+                                port_rt_info.update(self.__read_rt_info_common(attr))
+
+                    input_shape = shape_array([d if d != -1 else dynamic_dimension_value for d in input_shape])
+
+                    in_tensor_names = None
+                    if 'names' in port.attrib:
+                        in_tensor_names = port.attrib['names']
+
+                    # special attribute to pass information about operation input ports
+                    layer_attrs['restored_input_ports'].update({port_id: (input_shape, in_tensor_names, port_rt_info)})
             elif attr.tag == 'output':
                 output = attr
                 for port in output:
                     port_id = int(port.attrib['id'])
                     output_shape = []
+                    port_rt_info = {}
                     for dim in port:
                         if dim.tag == "dim":
                             output_shape.append(int(dim.text))
+                        if dim.tag == 'rt_info':
+                            for attr in dim:
+                                port_rt_info.update(self.__read_rt_info_common(attr))
 
                     output_shape = shape_array([d if d != -1 else dynamic_dimension_value for d in output_shape])
 
                     out_tensor_names = None
                     if 'names' in port.attrib:
                         out_tensor_names = port.attrib['names']
-
-                    layer_attrs['ports'].update({port_id: (output_shape, out_tensor_names)})
+                    # special attribute to pass information about operation input ports
+                    # NOTE: renaming of structure changing of this attribute may have big impact on tests
```

Comment:
renaming **OR** structure changing of this attribute may have big impact on tests

Reply:
Fixed.